test/analytics/fixtures/plugins/analytics_ftr_helpers/server/plugin.ts

```diff
@@ -45,34 +46,34 @@ export class AnalyticsFTRHelpers implements Plugin {
           query: schema.object({
             takeNumberOfEvents: schema.number({ min: 1 }),
             eventTypes: schema.arrayOf(schema.string()),
+            withTimeoutMs: schema.maybe(schema.number()),
+            fromTimestamp: schema.maybe(schema.string()),
           }),
         },
       },
       async (context, req, res) => {
-        const { takeNumberOfEvents, eventTypes } = req.query;
-
-        const events = await firstValueFrom(
-          events$.pipe(
-            filter((event) => {
-              if (eventTypes.length > 0) {
-                return eventTypes.includes(event.event_type);
-              }
-              return true;
-            }),
-            take(takeNumberOfEvents),
-            toArray(),
-            // Sorting the events by timestamp... on CI it's happening an event may occur while the client is still forwarding the early ones...
-            map((_events) =>
-              _events.sort(
-                (a, b) => new Date(a.timestamp).getTime() - new Date(b.timestamp).getTime()
-              )
-            )
-          )
-        );
-
+        const { takeNumberOfEvents, ...options } = req.query;
+        const events = await fetchEvents(events$, takeNumberOfEvents, options);
         return res.ok({ body: events });
       }
     );
+
+    router.get(
```

Comment:
Not sure what this API should be used for. Mind clarifying?

Reply:
Sorry! I just noticed this comment. It is called by the FTR helper (server-side implementation) in `test/analytics/services/kibana_ebt.ts`